addons/base_import/models.py

```diff
@@ -126,6 +134,56 @@ def get_fields(self, cr, uid, model, context=None,
         # TODO: cache on model?
         return fields
 
+    def _read_import_file(self, record, options):
+        rows = []
+        if record.file_type == 'text/csv':
```

Comment:
A dispatch dict (content_type -> handler function) or at least a dispatch method (takes content_type, returns handler method) would probably make sense, and make for easier overloading.